storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
Ah, indeed I rememberd the behavior around this the other way around. Ok, how about I just get rid of all the naked returns in this function anyhow, since naked returns are evil, as we can see here.